apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -1941,6 +1969,377 @@ export class LemonadePipeline implements BasePipeline {
+        return [];
+      }
+
+      const enabledFields = this.definition.options.offlineCheckin.fields;
+      const offlineTickets: PodboxOfflineTicket[] = [];
+
+      for (const atom of await this.db.load(this.id)) {
+        if (!atom.email) {
+          continue;
+        }
+
+        // If the user can check in tickets of this type
+        if (ticketMatchesCriteria(atom, checkerTicketCriteria)) {
+          offlineTickets.push({
+            id: atom.id,
+            eventId: atom.eventId,
+            eventName: this.lemonadeAtomToEventName(atom),
+            ticketName: this.lemonadeAtomToTicketName(atom),
+            attendeeEmail: enabledFields.attendeeEmail
+              ? atom.email.toLowerCase()
+              : undefined,
+            attendeeName: enabledFields.attendeeName ? atom.name : undefined,
+            checker: enabledFields.checker ? null : undefined,
+            is_consumed: atom.checkinDate instanceof Date
+          });
+        }
+      }
+
+      if ((this.definition.options.manualTickets ?? []).length) {
+        const manualCheckIns = await this.checkinDB.getByPipelineId(this.id);
+        for (const manualTicket of this.definition.options.manualTickets ??
+          []) {
+          // If the user can check in tickets of this type
+          if (ticketMatchesCriteria(manualTicket, checkerTicketCriteria)) {
+            const event = this.getEventById(manualTicket.eventId);
+            const product = this.getTicketTypeById(
+              event,
+              manualTicket.productId
+            );
+            const checkIn = manualCheckIns.find(
+              (checkIn) => checkIn.ticketId === manualTicket.id
+            );
+
+            offlineTickets.push({
+              id: manualTicket.id,
+              eventId: event.genericIssuanceEventId,
+              eventName: event.name,
+              ticketName: product.name,
+              attendeeEmail: enabledFields.attendeeEmail
+                ? manualTicket.attendeeEmail
+                : undefined,
+              attendeeName: enabledFields.attendeeName
+                ? manualTicket.attendeeName
+                : undefined,
+              checker: enabledFields.checker ? null : undefined,
+              is_consumed: checkIn?.timestamp instanceof Date
+            });
+          }
+        }
+      }
+
+      span?.setAttribute("offline_tickets_returned", offlineTickets.length);
+
+      return offlineTickets;
+    });
+  }
+
+  /**
+   * Stores ticket IDs that were checked in by an offline user for asynchronous
+   * check-in with the back-end service.
+   */
+  private async checkInOfflineTickets(
+    checkerEmail: string,
+    ticketIds: string[]
+  ): Promise<void> {
+    return traced(LOG_NAME, "checkInOfflineTickets", async (span) => {
+      tracePipeline(this.definition);
+
+      if (this.definition.options.offlineCheckin === undefined) {
+        throw new PCDHTTPError(
+          401,
```

Comment:
nit but i think `400` makes more sense for this type of error, here and elsewhere